src/Language/PureScript/TypeChecker.hs

```diff
@@ -281,9 +283,17 @@ typeCheckAll moduleName _ ds = traverse go ds <* traverse_ checkFixities ds
     warnAndRethrowWithPosition pos $ PositionedDeclaration pos com <$> go d
 
   checkFixities :: Declaration -> m ()
-  checkFixities (FixityDeclaration _ name (Just alias)) = do
+  checkFixities (FixityDeclaration _ name (Just (Left alias))) = do
     ty <- lookupVariable moduleName alias
     addValue moduleName (Op name) ty Public
+  checkFixities (FixityDeclaration _ name (Just (Right alias))) = do
+    env <- getEnv
+    case M.lookup alias (dataConstructors env) of
+      Nothing -> throwError . errorMessage $ UnknownDataConstructor alias Nothing
+      Just (_, _, ty, _) -> do
+        let v = Constructor alias
+        (_, ty') <- sndM (introduceSkolemScope <=< replaceAllTypeSynonyms) <=< instantiatePolyTypeWithUnknowns v $ ty
+        addValue moduleName (Op name) ty' Public
```

Comment:
This is basically just lifted from `infer`, do I actually need to do all this here, or would it be safe to just `addValue` the `ty`?